### ISSUE TITLE
Poses of SolidPrimitive and Mesh are treated differently when attaching objects

### ISSUE THREAD

**Issue by bit-pirate** [moveit/moveit_ros#306]
When attaching objects, it looks like the pose of SolidPrimitive is interpreted as being relative to the attached link, while the pose of Mesh is interpreted relative to the header's frame_id. Is this supposed to work like that? If yes, it should be mentioned in the message comments.

**Comment by isucan** [comment]
I just looked at the code again. I am pretty sure the poses are processed the same way. Are you sure you are setting the frame_id the same way? Meshes may also appear to be positioned differently because of the transform the CAD file was saved with.

**Comment by bit-pirate** [comment]
I'm using the mesh file from the household_objects_database. When adding them to the robot, I have their pose relative to "base_footprint", which I put in the header's frame_id of the CollisionObject. It works fine like that.

Afterwards I added a simple shape, aka SolidPrimitive. I passed in the same pose as for mesh, since I expected it needs to be relative to "base_footprint" as well. However, it turned out the pose is interpreted as being relative to the attached link. So, I had to manually adjust the pose for the SolidPrimitive.
